tracer/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -2587,13 +2587,14 @@ HRESULT CorProfiler::CallTarget_RewriterCallback(RejitHandlerModule* moduleHandl
         // Load the arguments directly (FastPath)
         for (int i = 0; i < numArgs; i++)
         {
-            reWriterWrapper.LoadArgument(i + (isStatic ? 0 : 1));
             const auto& argTypeFlags = methodArguments[i].GetTypeFlags(elementType);
             if (argTypeFlags & TypeFlagByRef)
             {
-                Logger::Warn("*** CallTarget_RewriterCallback(): Methods with ref parameters "
-                             "cannot be instrumented. ");
-                return S_FALSE;
+                reWriterWrapper.LoadArgument(i + (isStatic ? 0 : 1));
+            }
+            else
+            {
+                reWriterWrapper.LoadArgumentRef(i + (isStatic ? 0 : 1));
```

Comment:
Are these branches back to front? 🤔 if `argTypeFlags & TypeFlagByRef` is true, I was expecting `LoadArgumentRef()`? Of course the IL is a mystery to me, so I assume I'm wrong 😄

Reply:
If the argument is a ref, we use the normal `LDARG` (`LoadArgument`) because we already have the address, and the `CallTarget.Invoker` expect an address.

If the argument is a normal value (not ref) then we use `LDARGA` (`LoadArgumentRef`) to get the address of that value.

Reply:
Cool, thanks for clarifying 🙂